### ISSUE TITLE
Allow hot/live reloading

### ISSUE THREAD

**Issue by alexlemaire** [kaskadi/repo-sync-api#1]
**Is your feature request related to a problem? Please describe.**
If we send a synchronization request to the API itself or to another repository with a running process watching for file change, it will trigger a process restart before the complete synchronization is done (because once files are pulled, the watcher sees file changes and therefore restarts the process)

**Describe the solution you'd like**
It would be nice to have a way to reload the process only once the synchronization is complete. Maybe using a temporary folder to which we pull and then move all the folder content to the actual destination?

**Comment by alexlemaire** [comment]
Implemented, closing now.
